api/resolvers/embed.js

```diff
@@ -0,0 +1,48 @@
+import jsdom from 'jsdom'
+import crypto from 'crypto'
+import { EMBED_META_CACHE_DURATION_S } from '@/lib/constants'
+import { youtubeClipMetaFetchArgsSchema, validateSchema } from '@/lib/validate'
+
+export default {
+  Query: {
+    fetchEmbedMeta: async (parent, { provider, args }, { models, me }) => {
+      const hash = crypto.createHash('sha256').update(JSON.stringify({ provider, args })).digest('hex')
+
+      let meta = (await models.embedMeta.findUnique({
+        where: { hash }
+      }))?.meta
+
+      if (meta?.updatedAt && (Date.now() - meta.updatedAt.getTime()) / 1000 > EMBED_META_CACHE_DURATION_S) {
```

Comment:
Don't cache this forever, so that the system recovers at some point if youtube does breaking changes